catroid/src/org/catrobat/catroid/content/actions/IfLogicAction.java

```diff
@@ -48,7 +49,12 @@ protected void begin() {
 				return;
 			}
 			Double interpretation = ifCondition.interpretDouble(sprite);
-			ifConditionValue = interpretation.intValue() != 0 ? true : false;
+			String interpretationString = ifCondition.interpretString(sprite);
+			if (interpretationString.equals(R.string.brick_if_default_value)) {
+				ifConditionValue = true;
+			} else {
+				ifConditionValue = interpretation.intValue() != 0 ? true : false;
+			}
```

Comment:
Please always take a look at the ["Intention action icons"](https://www.jetbrains.com/idea/help/intention-actions.html) (yellow light bulb and highlighted code) of your written code. It automatically simplifies your code to
```ifConditionValue = interpretationString.equals(R.string.brick_if_default_value) || (interpretation.intValue() != 0);```

However, `R.string.brick_if_default_value` is an int. So you're comparing a String with an int which will always be false.